### PR TITLE
Cache choco packages to work around connectivity issues

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Description

This pull request includes changes to the `.github/workflows/reusable-test.yml` file, specifically in the `jobs:` section. The changes involve setting up a local cache for Chocolatey (choco) packages and adding a cache entry for any installed choco packages. The cache key is based on the hash of the workflow file, so any changes to the choco packages in the file will invalidate the cache.

Key changes:

* [`.github/workflows/reusable-test.yml`](diffhunk://#diff-e386584fe70c301924ab6a8d4656b54ef92ed1961e6eb6f77c155be3475615f6R112-R130): Added a new step to create a local folder for caching Chocolatey packages. The cache location is set using the `choco config set` command. This step is skipped if the `should_skip` output of the `skip_check` step is 'true'.
* [`.github/workflows/reusable-test.yml`](diffhunk://#diff-e386584fe70c301924ab6a8d4656b54ef92ed1961e6eb6f77c155be3475615f6R112-R130): Added another step to cache the Chocolatey packages. The cache uses the `actions/cache` action with a specific commit hash. The cache key is generated using the `hashFiles` function with the workflow file as input. This step is also skipped if the `should_skip` output of the `skip_check` step is 'true'.

## Testing

CI/CD

## Documentation

No.

## Installation

No.

### CODE REVIEW

File: .github/workflows/reusable-test.yml [hunk 1]
```diff
@@ -103,15 +103,33 @@ jobs:
         submodules: 'recursive'
         ref: ${{ github.event.workflow_run.head_branch }}
 
+
```
Comment (dthaler) [inline .github/workflows/reusable-test.yml:106]: Why the extra blank line?
```suggestion
```

File: .github/workflows/reusable-test.yml [hunk 2]
```diff
@@ -103,15 +103,33 @@ jobs:
...
     # Perform shallow checkout for self-hosted runner.
     - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b
       if: (inputs.environment == 'ebpf_cicd_tests_ws2019' || inputs.environment == 'ebpf_cicd_tests_ws2022' || inputs.environment == 'ebpf_cicd_perf_ws2022') && (steps.skip_check.outputs.should_skip != 'true')
       with:
         ref: ${{ github.event.workflow_run.head_branch }}
 
+    # Check if .github/workflows/reusable-test.yml exists locally
```
Comment (dthaler) [inline .github/workflows/reusable-test.yml:113]:
```suggestion
    # Check if .github/workflows/reusable-test.yml exists locally.
```
Style consistency nit.

Comment (dthaler) [reply .github/workflows/reusable-test.yml:113]: Also, when would it not exist locally?

File: .github/workflows/reusable-test.yml [hunk 3]
```diff
@@ -103,15 +103,33 @@ jobs:
...
+
+    # Check out just this file if code hasn't been checked out yet.
+    - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b
+      if: (steps.check_reusable_test_locally.outputs.files_exists != 'true') && (steps.skip_check.outputs.should_skip != 'true')
+      with:
+        sparse-checkout: |
+          .github/workflows/reusable-test.yml
+        sparse-checkout-cone-mode: false
+
     - name: Setup choco cache folder
```
Comment (dthaler) [inline .github/workflows/reusable-test.yml:130]:
```suggestion
    - name: Set up choco cache folder
```

**Review by dthaler — APPROVED** [review]

**Review by dv-msft — APPROVED** [review]

**Pull request merged**